### PR TITLE
Serve on Yext Sites

### PR DESCRIPTION
**Pull request by cea2aj**
Serve on yext sites

This adds default configuration to support yext sites. It works for both full production builds and for live preview. Currently the page is being served by the dev branch, but it will be available for develop and main once it's merged into those branches. I opted for `npm install` for the `installDepsCmd` so that the package.json file can be modified in live preview and the package-lock.json can be updated as necessary. For the `buildSetupCmd`, I went with `npm ci` because it's faster and ensures we get a clean build based purely on the package-lock.json file.

J=SLAP-1841
TEST=manual

Add the repo to yext sites and view the page being served at https://devyextsites-react--site--search--starter-slapshot-pagescdn-com.preview.pagescdn.com/. Also load the experience in the code editor and see the page work with live preview. Modify a file and see the file reflected in the page after refreshing the page

### CODE REVIEW

File: sites-config/ci_config.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "artifactStructure": {
+      "assets": [
+          {
+              "root": "build"
+          }
+      ],
+      "hbsTemplates": []
+  },
+  "base_image_tag": "node-14",
```
Comment (oshi97) [inline sites-config/ci_config.json:10]: can we use node 16? I think I tried 17 locally a month-ish and it didn't work, though if it did that might be even better?

File: sites-config/ci_config.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "dependencies": {
+      "installDepsCmd": "npm install",
+      "requiredFiles": []
+  },
+  "buildArtifacts": {
+      "build_setup_cmd": "npm install",
```
Comment (oshi97) [inline sites-config/ci_config.json:16]: is installDepsCmd not run before buildArtifacts?

Comment (cea2aj) [reply sites-config/ci_config.json:16]: ah, you're right that we should be able to rely on the install from the `installDepsCmd`. I'll try that

File: sites-config/ci_config.json (hunk 1 above)
Comment (cea2aj) [reply sites-config/ci_config.json:10]: I can give it a shot

File: sites-config/ci_config.json (hunk 2 above)
Comment (cea2aj) [reply sites-config/ci_config.json]: turns out it supports camel case for everything!

Comment (oshi97) [reply sites-config/ci_config.json]: oh wow they support both that's pretty crazy O.o

Comment (cea2aj) [reply sites-config/ci_config.json]: When I run the install just in the `build_setup_cmd`, live preview breaks, and if I run the install just in the `install_deps_cmd`, the build breaks, so it looks like we need it for both which is what Theme 1.0 sites do

File: sites-config/ci_config.json (hunk 1 above)
Comment (cea2aj) [reply sites-config/ci_config.json]: node 17 is working!

File: sites-config/ci_config.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  "dependencies": {
+      "installDepsCmd": "npm install",
+      "requiredFiles": []
+  },
+  "buildArtifacts": {
+      "buildSetupCmd": "npm ci",
+      "buildCmd": "npm run build"
+  },
+  "livePreview": {
+      "serveCmd": "export PORT=8080 && npm run start"
```
Comment (oshi97) [inline sites-config/ci_config.json:20]: this a super nit but I think you can just do `PORT=8080 npm start` (or npm run start, just dont think we want the export PORT)

File: sites-config/ci_config.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "artifactStructure": {
+      "assets": [
+          {
```
Comment (oshi97) [inline sites-config/ci_config.json:4]: ahh another nit (and I should have noticed this earlier) but I think the spacing is inconsistent? think the vscode `> Format Document` should fix stuff

**Review by oshi97 — APPROVED** [review]

**Pull request merged**